### ISSUE TITLE
Dashboard: Yesterday items show up in today

### ISSUE THREAD

**Issue by dwertheimer** [NotePlan/plugins#497]
Today is 11/21 but 11/20 items are showing up in the today section.

This only happens if I have the "Show 'Yesterday' Section" setting turned on.

<img width="505" alt="Screen Cap 2023-11-21 at 07 00 18@2x" src="https://github.com/NotePlan/plugins/assets/8949588/a9ad7727-befb-46c0-a4cb-1b0773d1293d">

I see this in the log
2023-11-21 06:57:26 | DEBUG | getOpenItemParasForCurrentTimePeriod :: Processing 20231121.md which has 0 paras
...
2023-11-21 06:57:26 | DEBUG | getOpenItemParasForCurrentTimePeriod :: Processing 20231120.md which has 48 paras

also can we pls turn down some of the logging?

<img width="593" alt="Screen Cap 2023-11-21 at 07 06 05@2x" src="https://github.com/NotePlan/plugins/assets/8949588/ec2d5efb-96c6-4b71-a24a-5a461d0762e5">

**Comment by jgclark** [comment]
Dates: OK, I will look. It'll be from a pesky non-locale aware helper function.

Logging: I keep logging up when I've been making bigger changes. I will tone down again once things stabilise. You can always turn the logging down yourself for now!

**Comment by jgclark** [comment]
Hmm, perhaps I spoke too soon. I've never seen this happen, and I can't reproduce it.
Relevant code is:
```js
    const today = new moment().toDate()
    let currentDailyNote = DataStore.calendarNoteByDate(today, 'day')
    if (currentDailyNote) {
      const thisFilename = currentDailyNote?.filename ?? '(error)'
      const [combinedSortedParas, sortedRefParas] = await getOpenItemParasForCurrentTimePeriod("day", currentDailyNote, config)
...

and that helper starts (where timePeriodNote = currentDailyNote):

...
    logDebug('getOpenItemParasForCurrentTimePeriod', `Processing ${timePeriodNote.filename} which has ${String(timePeriodNote.paragraphs.length)} paras`)
    parasToUse = timePeriodNote.paragraphs
    logInfo('getOpenItemParasForCurrentTimePeriod', `Got ${parasToUse.length} parasToUse (after ${timer(startTime)})`)
```
I've been switching everything to use moment library over time, particularly because it deals in locale time by default, unlike JS built-ins. So I can't yet see what might be going wrong.

**Comment by jgclark** [comment]
I'm turning down logging quite a bit for 0.7.4. 
But on the main point about dates: are you still seeing this problem?  If not, please close.

**Comment by dwertheimer** [comment]
@jgclark I turned on the "yesterday" pref and I got this:

![Screen Cap 2023-12-11 at 06 48 42@2x](https://github.com/NotePlan/plugins/assets/8949588/933ec211-6fe4-4e90-97e9-2af06513d61e)

Yesterday's overdue task shows up in today.

Will DM you the log.

**Comment by jgclark** [comment]
This later problem has been fixed in 1.0.0-a4, so closing this.
